daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -291,6 +305,18 @@ class VariableMap private(vTable: Map[GlobalQName, ArrayBuffer[VariableInstance]
     new VariableMap(table)
   }
 
+  def topLevelInstances(): VariableMap = {
+    val table = vTable.map { case (k: GlobalQName, abuf: ArrayBuffer[VariableInstance]) => {
+      val newBuf = new ArrayBuffer[VariableInstance]()
+      if (abuf.length > 1)
+        newBuf.append(abuf(abuf.length - 2))
+      newBuf.append(abuf.last)
+      (k, newBuf)
```

Comment:
Ah, this was something that should have been removed.  It was necessary to support the self-referencing NVI statements, but since we aren't allowing those any more it ls no longer needed.